contracts/src/locker/token_locker.cairo

```diff
@@ -196,9 +204,18 @@ mod TokenLocker {
             let mut token_lock = self.locks.read(lock_id);
 
             // Update user locks
-            self.remove_user_lock(lock_id, token_lock.owner);
+            let mut user_locks = self.user_locks.read(token_lock.owner);
+            // Update token locks
+            let mut token_locks = self.token_locks.read(token_lock.token);
+            // Removing user lock
+            self.remove_lock_from_list(lock_id, user_locks);
+            // Removing token lock
+            self.remove_lock_from_list(lock_id, token_locks);
+
             let mut new_owner_locks: List<u128> = self.user_locks.read(new_owner);
             new_owner_locks.append(lock_id);
+            let mut new_token_locks : List<u128> =  self.token_locks.read(token_lock.token);
```

Comment:
ah, ok I get it: this is redundant, you remove the lock to put it back right after in the same list

Reply:
True true